# Surviving the Titanic

This contains a few exercises focused on using data frames to begin looking at data.  It uses the titanic.csv dataset.

Relevant textbook section: [7.1 - An Introduction to Working with Dataframes](https://snakebear.science/07-Pandas/Dataframes.html)

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# For slightly nicer charts
plt.rcParams['figure.figsize'] = [10, 6]
plt.rcParams['figure.dpi'] = 150

***
Now we read a csv file of passengers of the titanic and assign it to a variable called `df`. Each row in the dataset represents one passenger on the titanic. Each row contains information about the passenger and whether they survived. The columns are labelled as follows: 
* 'Survived'
* 'Pclass'
* 'Name'
* 'Sex'
* 'Age'
* 'Siblings/Spouses Aboard'
* 'Parents/Children Aboard'
* 'Fare'

The dataset comes from here http://web.stanford.edu/class/archive/cs/cs109/cs109.1166/problem12.html

In [5]:
df = pd.read_csv("titanic.csv")

Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'Siblings/Spouses Aboard',
       'Parents/Children Aboard', 'Fare'],
      dtype='object')

At this point all we know is that we've succesfully read a csv file. Let's find out some more about the data. First, let's find out how much data there is in the file. 

In [6]:
df.shape

(887, 8)

`shape` returns two numbers that represent the number of rows and the number of columns in your dataframe. The titanic data set has 887 rows (each representing one passenger) and 8 columns. 
It might be a good idea to get a look at our dataframe but 887 rows seems like a bit much to look at all at once. We can look at just part of the dataframe by using `head'.

In [7]:
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

That looks pretty good. Now what we are interested in is what might predict whether someone survived the titanic.
First let's find out how many people lived. 

In [15]:
df['Survived'].value_counts()

0    545
1    342
Name: Survived, dtype: int64

What we've done here is apply the `value_counts` method to the 'Survived' column of the dataframe. It looks like there was 342 survivors.
Now its your turn used the same method to look at the counts of Passenger Class and Sex. 

In [20]:
# enter your code here

What we are doing right now is getting to know out data. Let's get to know if just a bit better by looking at histograms of Age and Fare.

In [21]:
# enter your code here

***
Okay, now that you are a bit more familiar with your data back to the question at hand. What predicts surviving this disaster? Use the tools we've presented in the course to explore this question. Approach it any way you like. Remember though, whatever you come up with will probably need to compare survivors to those that didn't. (For example, we could look at how Fare varied by Passenger Class but that wouldn't tell us much about surviving). 

In [21]:
# enter your code here